dtp/src/contracts.rs

```diff
@@ -0,0 +1,138 @@
+        tx_receipt.block_number.unwrap()
+    );
+
+    Ok(())
+}
+
+/// increment Counter number
+/// NOTE: signer needed as it incurs gas fees.
+pub(crate) async fn counter_increment(
+    client: Arc<Provider<Http>>,
+    counter_address: Address,
+    signer: Wallet<SigningKey>,
+    chain_id: u64,
+) -> eyre::Result<()> {
+    // create a middleware client with signature from signer & provider
+    let client_middleware = SignerMiddleware::new(client.clone(), signer.with_chain_id(chain_id));
+
+    // clone the client (if multiple use)
+    let client_middleware = Arc::new(client_middleware);
+
+    // get a contract
+    let counter = Counter::new(counter_address, client_middleware);
+
+    // send a transaction with setter function
+    let tx_receipt = counter
+        .increment()
+        .send()
+        .await
+        .expect("Failure in getting pending tx")
+        .await?
+        .expect("Failure in \'increment\' of Counter contract");
+    info!(
+        "\'{}\' increment number, which incurred a gas fee of \'{} TSSC\', has a tx hash: \'{:?}\', indexed at #{} in block #{}.\n",
+        tx_receipt.from,
+        get_gas_cost(&client, &tx_receipt).await?,
+        tx_receipt.transaction_hash,
+        tx_receipt.transaction_index,
+        tx_receipt.block_number.unwrap()
+    );
+
+    Ok(())
+}
+
+/// Load contract: `setArray` method
+/// NOTE: signer needed as it incurs gas fees.
+pub(crate) async fn load_set_array(
+    client: Arc<Provider<Http>>,
+    load_address: Address,
+    signer: Wallet<SigningKey>,
+    chain_id: u64,
+    max_load_count_per_block: u16,
+) -> eyre::Result<()> {
+    // create a middleware client with signature from signer & provider
+    let client_middleware = SignerMiddleware::new(client.clone(), signer.with_chain_id(chain_id));
+
+    // clone the client (if multiple use)
+    let client_middleware = Arc::new(client_middleware);
+
+    // get a contract
+    let load = Load::new(load_address, client_middleware);
+
+    // TODO: Here, `count` can be abstracted out as CLI parameter with default value set as may be `1000`
+    // considered the highest possible count per block for now.
+    let count = max_load_count_per_block;
+
+    // send a transaction with setter function
+    let tx_receipt = load
+        // to try out with its different values.
+        // The max. `count` possible in `setArray` method of Load contract is 2650. Above this count value,
+        // the gas cost exceeds 60 M per block (as set for Subspace EVM domain).
+        .set_array(U256::from(count))
+        .send()
+        .await
+        .expect("Failure in getting pending tx")
+        .await?
+        .expect("Failure in \'setArray\' method of Load contract");
+
+    info!(
+        "\'{}\' set Array with count {}, which incurred a gas fee of \'{} TSSC\', has a tx hash: \'{:?}\', indexed at #{} in block #{}.\n",
+        tx_receipt.from,
+        count,
+        get_gas_cost(&client, &tx_receipt).await?,
```

Comment:
I think this call is wasteful and that we don't really need this  info.

Reply:
This info is although showing as log when run in debug mode, otherwise not. Do you still want me to remove this `get_gas_cost` or `info!(...)` function call altogether.

Reply:
wouldn't `debug!` be what you are looking for here?

Reply:
Initially, I was dubious about their usage initially. But when I learnt more about the differences in usage, the current details are not too verbose. So, used `info!` instead of `debug!`.

This is the info I found as their difference when to use:
<details>
<summary>See the differences:</summary>
Example:

Consider an application that processes user transactions:

- You might use `log::info!` to indicate that the application has started processing a batch of transactions.
- You would use `log::debug!` to log the details of each transaction in the batch, which might include intermediate states, specific values being processed, or decisions made by the transaction processing logic.

</details>

Reply:
My concern is the async calls that happen within `get_gas_cost`, not so much the verbosity.

Reply:
[commit](https://github.com/subspace/domain-transaction-producer/pull/5/commits/f0eb6cd01dd1078bde88441cdbc6199e7fccb11d)